pkg/workflow/task/bootstrap/client.go

```diff
@@ -0,0 +1,46 @@
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+// ClientOptionsRetriever supplies bootstrap cluster options to be used in bootstrap cluster
+// creation.
+type ClientOptionsRetriever interface {
+	GetCreateBootstrapClusterOptions(*cluster.Spec) CreateBootstrapClusterOptions
+}
+
+// Client provides behavior for interacting with a bootstrap cluster.
+type Client interface {
+	// CreateBootstrapCluster creates a bootstrap cluster with the given name and options.
+	CreateBootstrapCluster(ctx context.Context, name string, opts CreateBootstrapClusterOptions) error
+
+	// DeleteBootstrapCluster deletes a bootstrap cluster identified with the given name.
+	DeleteBootstrapCluster(ctx context.Context, name string) error
+
+	// GetKubeconfig retrieves an in-memory kubeconfig for the bootstrap cluster identified by
+	// name.
+	GetKubeconfig(ctx context.Context, name string) ([]byte, error)
+
+	// ClusterExists determines if a bootstrap cluster identified by name exists.
+	ClusterExists(ctx context.Context, name string) (bool, error)
+}
+
+// CreateBootstrapClusterOptions defines the options available for bootstrap cluster creation.
+type CreateBootstrapClusterOptions struct {
```

Comment:
Curious why you would define this here as opposed to in `pkg/bootstrap` or similar.

This creates dependencies "retriever" -> `pkg/workflow/task/bootstrap` and "client" -> `pkg/workflow/task/bootstrap` and I'm not sure that's a good thing.

Although there is no other usecase for the bootstrap client/retriever than to be used from these tasks, it seems a bit odd than they depend on this package, which is just an orchestrator.

Reply:
I'm not sure what we want yet. With the other changes in this PR, the `bootstrap.Bootstrapper` is redundant and disappears. If we put the options struct in `boostrapper` it only defines that type making that package feel rather unnecessary.

This would be a tradeoff that yields colocated code at the expense of an odd looking import statement in things populating the options.

Reply:
Another thought. It might be worth simplifying the options first to extract as much provider agnostic configuration out of providers and see whats left.